### PR TITLE
Pre 0.6.4 Updates 

### PR DESCRIPTION
**Pull request by chernser**
## Summary
- Version updates 
- Changelog updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,46 @@
 ## Latest
 
+## 0.6.4
+
+### Important
+- Following components will be deprecated and archived in next release:
```
Comment (mshustov) [inline CHANGELOG.md:6]: Have we confirmed the removal in the next release?  If no, I'd move the deprecation warning at the top to ensure it's visible to everyone reading the file

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,46 @@
...
+  - clickhouse-cli-client
+  - clickhouse-grpc-client
+- No more builds for non-lts Java versions - no more Java 9 release builds.
```
Comment (mshustov) [inline CHANGELOG.md:9]: Isn't it a breaking change for users with the old Java version? If so, should we tag the release as `0.7.0`?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,46 @@
...
+- Lowest supported Java version will be 11.
+  - Java 11 support will be ended before the end of 2023. 
+  - It is recommended to use Java 21. 
+
+### New Features 
+- [Client-V1/Apache HTTP] More configuration parameters for connection management. Useful for tuning performance.
+(https://github.com/ClickHouse/clickhouse-java/pull/1771)
+    - com.clickhouse.client.config.ClickHouseClientOption#CONNECTION_TTL - to configure connection time-to-live
+    - com.clickhouse.client.http.config.ClickHouseHttpOption#KEEP_ALIVE_TIMEOUT - to configure keep-alive timeout
+    - com.clickhouse.client.http.config.ClickHouseHttpOption#CONNECTION_REUSE_STRATEGY - to configure how connection
```
Comment (mshustov) [inline CHANGELOG.md:19]: The setting name is self-descriptive. We'd rather focus on the meaning of the setting and possible configuration value. You can even use the existing comments from javadoc https://github.com/ClickHouse/clickhouse-java/blob/82f8b2125ab3f688bcfc9c44a2112394cf2aebb0/clickhouse-http-client/src/main/java/com/clickhouse/client/http/config/ClickHouseHttpOption.java#L93-L96

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,46 @@
...
+- [Client-V2] Connection pool configuration https://github.com/ClickHouse/clickhouse-java/pull/1766
+    - com.clickhouse.client.api.Client.Builder.setConnectionRequestTimeout - to configure connection request timeout.
+Important when there are no connections available in the pool to fail fast. 
+    - com.clickhouse.client.api.Client.Builder.setMaxConnections - total max connections per route.
+    - com.clickhouse.client.api.Client.Builder.setConnectionTTL - to limit connection live ignoring keep-alive from server.
+    - com.clickhouse.client.api.Client.Builder.setConnectionReuseStrategy - to configure how connections are used.
+Select FIFO to reuse connections evenly or LIFO (default) to reuse the most recently active connections.
+- [Client-V2] All operations are now executed in calling thread to avoid extra threads creation. 
+Async operations can be enabled by `com.clickhouse.client.api.Client.Builder.useAsyncRequests` (https://github.com/ClickHouse/clickhouse-java/pull/1767)
+- [Client-V2] Spring Demo Service as usage example (https://github.com/ClickHouse/clickhouse-java/pull/1765)
```
Comment (mshustov) [inline CHANGELOG.md:32]: Should we group the code and docs changes into relevant sections? it would help to skim through the file

**Review by mshustov — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (chernser) [reply CHANGELOG.md:6]: Moved to deprecations section.

File: CHANGELOG.md (hunk 2 above)
Comment (chernser) [reply CHANGELOG.md:9]: Discussed. It will be in deprecations warning and will be removed in 0.7.0 release.

File: CHANGELOG.md (hunk 4 above)
Comment (chernser) [reply CHANGELOG.md:32]: Moved to documentation section.

File: CHANGELOG.md (hunk 3 above)
Comment (chernser) [reply CHANGELOG.md:19]: Updated.

**Pull request merged**